src/pages/index.js

```diff
@@ -0,0 +1,380 @@
+const profileDescription = document.querySelector(".profile__description");
+const profileName = document.querySelector(".profile__name");
+
+//Form Elements
+const editModal = document.querySelector("#edit-profile-modal");
+const editFormElement = editModal.querySelector(".modal__form");
+const editProfileModal = document.querySelector("#edit-profile-modal");
+const closeButtons = document.querySelectorAll(".modal__close");
+const editModalNameInput = editModal.querySelector("#profile-name-input");
+const editModalDescriptionInput = editModal.querySelector(
+  "#profile-description-input"
+);
+
+const cardModal = document.querySelector("#add-card-modal");
+const cardForm = cardModal.querySelector(".modal__form");
+const cardSubmitButton = cardModal.querySelector(".modal__button");
+const cardModalCloseButton = cardModal.querySelector(".modal__close");
+const descriptionInput = cardModal.querySelector("#add-card-name-input");
+const cardLinkInput = cardModal.querySelector("#add-card-link-input");
+const settings = {
+  inactiveButtonClass: "button--disabled", // Ensure this class exists in your CSS
+};
+
+const editAvatarForm = document.getElementById("edit-avatar-form");
+const SaveButton = document.querySelector(".modal__button_save");
+console.log("SaveButton:", SaveButton);
+
+//avatar form element //
+const avatarModal = document.querySelector("#avatar-modal");
+const avatarForm = avatarModal.querySelector(".modal__form");
+const avatarSubmitButton = avatarModal.querySelector(".modal__button");
+console.log();
+const avatarModalCloseButton = avatarModal.querySelector(".modal__close");
+const avatarInput = avatarModal.querySelector("#profile-avatar-input");
+const profileAvatarContainer = document.querySelector(
+  ".profile__avatar-container"
+);
+const avatarImage = document.querySelector(".profile__avatar");
+
+// delete form elements
+const deleteModal = document.querySelector("#delete-modal");
+const deleteform = deleteModal.querySelector(".modal__form");
+const modalCancelButton = document.querySelector(".modal__button_cancel"); // Cancel button
+const modalDeleteButton = document.querySelector(".modal__button_delete");
+
+//select the modal
+const previewModal = document.querySelector("#preview__modal");
+const previewModalImageElement = previewModal.querySelector(".modal__image");
+const previewModalCaptionElement =
+  previewModal.querySelector(".modal__caption");
+const previewModalCloseButton = previewModal.querySelector(".modal__close");
+
+//Card related elements
+const cardTemplate = document.querySelector("#card-template");
+const cardsList = document.querySelector(".cards__list");
+
+let selectedCard, selectedCardId;
+
+function handleDeleteSubmit(evt) {
+  console.log("deleting card with ID", selectedCardId);
+  evt.preventDefault();
+  setButtonText(modalDeleteButton, true, "Delete", "Deleting...");
+
+  api
+    .deleteCard(selectedCardId)
+    .then(() => {
+      setButtonText(modalDeleteButton, false, "Delete");
```

Comment:
You need to **only** return the **default** button text in the `finally` block, so that the button could get the text regardless of whether the request succeeds or fails. 

**Please, correct it everywhere in the code**